docs/decisions/0003-lightweight-extension-points.rst

```diff
@@ -0,0 +1,202 @@
+            # .. filter_type: org.openedx.learning.ora.submission_view.render.started.v1
+            context, path = ORASubmissionViewRenderStarted.run_filter(context, path)
+        except ORASubmissionViewRenderStarted.RenderInvalidTemplate as exc:
+            context, path = exc.context, exc.template_name
+
+This implementation will allow us to modify what's rendered to the student, via the view context and template, for cases when needed. For example, some third-party services need acknowledgment before receiving users' information.
+
+The second extension point will be an Open edX Event. The event payload should contain enough information for later processing; in this case, we'll the following event definition:
+
+
+..  code-block:: python
+
+    attr.s(frozen=True)
+    class ORASubmissionData:
+        """
+        Attributes defined to represent event when a user submits an ORA assignment.
+
+        Arguments:
+            id (str): identifier of the ORA submission.
+            file_downloads (List[dict]): list of related files in the ORA submission. Each dict
+                contains the following keys:
+                    * download_url (str): URL to download the file.
+                    * description (str): Description of the file.
+                    * name (str): Name of the file.
+                    * size (int): Size of the file.
+        """
+        id = attr.ib(type=str)
+        file_downloads = attr.ib(type=List[dict], factory=list)
+
+    # .. event_type: org.openedx.learning.ora.submission.created.v1
+    # .. event_name: ORA_SUBMISSION_CREATED
+    # .. event_description: Emitted when a new ORA submission is created
+    # .. event_data: ORASubmissionData
+    ORA_SUBMISSION_CREATED = OpenEdxPublicSignal(
+        event_type="org.openedx.learning.ora.submission.created.v1",
+        data={
+            "submission": ORASubmissionData,
+        },
+    )
+
+The event will be sent `after a student submits a response to the assessment`_ so it has access to the student's submission key data:
+
+
+..  code-block:: python
+
+    @staticmethod
+    def send_ora_submission_created_event(submission: dict) -> None:
+        """
+        Send an event when a submission is created
+        Args:
+            submission (dict): The submission data
+        """
+        from openassessment.xblock.openassessmentblock import OpenAssessmentBlock
+
+        file_downloads = OpenAssessmentBlock.get_download_urls_from_submission(
+            submission
+        )
+        ORA_SUBMISSION_CREATED.send_event(
+            submission=ORASubmissionData(
+                id=submission.get("uuid"),
+                file_downloads=file_downloads,
+            )
+        )
+
+     # Sent after the submission
+     self.send_ora_submission_created_event(submission)
+
+This event will allow us to act after a submission is made based on the data sent.
+
+Consequences
+************
```

Comment:
Note to author: we might need adding a little more specifications about Turnitin? Or the _external services_ generic references are enough?

Reply:
I think that keeping it generic would be best since we could use any other third-party tool through the extension points.

Reply:
Agree. Thanks!